### PR TITLE
Add docs and example for formatted cell components

### LINKED ISSUE #228: Docs request for table cell format components
We have some sweet formatting components that are useful in table cells, but no docs rendered in the docs site.

https://github.com/palantir/blueprint/tree/master/packages/table/src/cell/formats

### PR DESCRIPTION
**Pull request by themadcreator**
Adds TableFormatsExample to docs.
Adds ITruncatedFormatProps and IJSONFormatProps interfaces to docs.

#### Checklist
- [x] Updated documentation

### CODE REVIEW

**Comment by themadcreator** [comment]
https://1199-71939872-gh.circle-artifacts.com/0/home/ubuntu/blueprint/packages/docs/dist/index.html#components.table-js.formatted

File: packages/table/examples/tableFormatsExample.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    ["+11:30", 11.5, "Pacific/Norfolk"],
+    ["+12:00", 12.0, "Pacific/Auckland"],
+    ["+12:45", 12.75, "Pacific/Chatham"],
+    ["+13:00", 13.0, "Pacific/Tongatapu"],
+    ["+14:00", 14.0, "Pacific/Kiritimati"],
+].map((arr) => {
+    return {
+        name: arr[2],
+        offsetMsec: (arr[1] as number) * 60 * 60 * 1000 + LOCAL_TIMEZONE_OFFSET_MSEC,
+        offsetString: arr[0],
```
Comment (giladgray) [inline packages/table/examples/tableFormatsExample.tsx]: perfect opportunity to deconstruct function arg:

```js
([offsetString, offsetHour, name]) => ({
    name,
    offsetMsec: offsetHour * 60 * 60 * 1000 + LOCAL_TIMEZONE_OFFSET_MSEC,
    offsetString, 
})
// note the enclosing parens so it's an implicit object return
```

File: packages/table/src/_docs.scss [hunk 2]
```diff
@@ -129,21 +129,33 @@ Loading states
 When fetching or updating data, it may be desirable to show a loading state. The Table components
 provide fine-grain loading control of loading row headers, column headers, or individual cells.
 
+`Cell`s expose a `loading` prop for granular control of which cells should show a loading state.
+Below is a table of the largest potentially hazardous asteroid (based on absolute magnitude)
+discovered in a given year. Try selecting a different preset loading configuration.
+
+@react-example CellLoadingExample
```
Comment (giladgray) [inline packages/table/src/_docs.scss]: 👍  nice move

File: packages/table/src/_docs.scss [hunk 3]
```diff
@@ -245,3 +257,37 @@ Weight: 5
...
+Styleguide components.table-js.api.truncated-format
+*/
+
+/*
+JSONFormat
+
+Wrap your javascript object cell contents with a `JSONFormat` component like so:
+
+```tsx
+const content = {any: "javascript variable", even: [null, "is", "okay", "too"]};
```
Comment (giladgray) [inline packages/table/src/_docs.scss]: spaces inside braces please

**Pull request merged**